File: golang/array_string/max_profit.go
package main

import "math"

func maxProfit(prices []int) int {
	minPrice := math.MaxInt32

	maxProfit := 0

	for _, price := range prices {
		if price < minPrice {
			minPrice = price
		}

		profit := price - minPrice

		if profit > maxProfit {
			maxProfit = profit
		}
	}

	return maxProfit
}

File: java/src/main/java/com/sanedge/interview/array_string/MaxProfit.java
package com.sanedge.interview.array_string;

public class MaxProfit {

    public int maxProfit(int[] prices) {
        int minPrice = Integer.MAX_VALUE;

        int maxProfit = 0;

        for (int price : prices) {
            if (price < minPrice) {
                minPrice = price;
            }

            int profit = price - minPrice;

            if (profit > maxProfit) {
                maxProfit = profit;
            }
        }

        return maxProfit;
    }
}
